fc_util/src/arg_parser.rs

```diff
@@ -0,0 +1,545 @@
+    }
+}
+
+#[derive(Default)]
+pub struct ArgParser<'a> {
+    pub args_matches: Vec<ArgMatch<'a>>,
+}
+
+impl<'a> ArgParser<'a> {
+    // Check if the argument at `index` position was provided with the right kind of
+    // value and also if there are not other conflicts.
+    fn check_conflicts(
+        &self,
+        argument_info: &ArgumentInfo<'static>,
+        trimmed_args: &[&str],
+        index: usize,
+        args: &[String],
+    ) -> Result<()> {
+        if argument_info.takes_value {
+            if let Some(arg_value) = args.get(index + 1) {
+                if arg_value.starts_with("--")
+                    && trimmed_args.contains(&arg_value.trim_start_matches("--"))
+                {
+                    return Err(Error::MissingValue(argument_info.name.to_string()));
+                }
+                if let Some(arg) = args.get(index + 2) {
+                    if !arg.starts_with("--") {
+                        return Err(Error::UnexpectedArgument(arg.clone()));
+                    }
+                }
+                if self
+                    .args_matches
+                    .iter()
+                    .any(|x| x.name == argument_info.name)
+                {
+                    return Err(Error::RepeatedArgument(argument_info.name));
+                }
+            } else {
+                return Err(Error::MissingValue(argument_info.name.to_string()));
+            }
+        } else {
+            if let Some(arg) = args.get(index + 1) {
+                if !arg.starts_with("--") {
+                    return Err(Error::UnexpectedArgument(arg.clone()));
+                }
+            }
+            if self
+                .args_matches
+                .iter()
+                .any(|x| x.name == argument_info.name)
+            {
+                return Err(Error::RepeatedArgument(argument_info.name));
+            }
+        }
+        if let Some(arg_name) = argument_info.requires {
+            if !trimmed_args.contains(&arg_name) {
+                return Err(Error::MissingArgument(arg_name));
+            }
+        }
+        Ok(())
+    }
+
+    // Extracts an argument's value or returns a specific error if the argument is missing.
+    pub fn get_value(&self, arg_name: &'static str) -> Result<String> {
+        if let Some(arg) = self.args_matches.iter().find(|x| x.name == arg_name) {
+            return Ok(arg.value.clone());
+        }
+        Err(Error::MissingArgument(arg_name))
+    }
+
+    // Populates `args_matches` field of ArgParser.
+    pub fn add_matched_args(
+        &mut self,
+        args: &'a [String],
+        args_info: &[ArgumentInfo<'static>],
+    ) -> Result<()> {
+        let seccomp_values = ["0", "1", "2"];
+        let trimmed_args: Vec<&str> = args
+            .iter()
+            .filter(|&x| x.starts_with("--"))
+            .map(|x| x.trim_start_matches("--"))
+            .collect();
+
+        if trimmed_args.contains(&"help") {
```

Comment:
This logic seems a bit problematic. I think we should print this only if the input is `firecracker help`. Otherwise if for example the input is `firecracker --exec-file help`, then `help` may be the name of the file.

Reply:
Here I am searching for `help` in the `trimmed_args` list, which contains the initial arguments that started with `--` (with `--` also removed), so the search is actually for `--help`. I was thinking to print the help information regardless of  `--help` position.

Reply:
I see. Ok.